lib/assert.js

```diff
@@ -271,8 +271,14 @@ function innerDeepEqual(actual, expected, strict, memos) {
       position: 0
     };
   } else {
-    if (memos.actual.has(actual)) {
-      return memos.actual.get(actual) === memos.expected.get(expected);
+    // We prevent up to two set.has(x) calls by directly retrieving the value
+    // and checking for undefined.
```

Comment:
It should say `map.has(x)` and not `set`. The only values in the map are numbers as it is a internal value.